### DISCUSSION TITLE
Capture original file name to another column ?

### DISCUSSION THREAD

**Discussion by HeartlandTechie** [filamentphp/filament#1434] [Help]
Any ideas or direction on extending a resource to give me a column with the original file name for an upload?  Renaming the file isn't a bad practice, but I really need to know what the original name was too.   Thanks!

**Comment by underdpt** [comment]
I've been dealing with this for a few hours and find no way except using `preserveFilenames()` which is suboptimal.

Could we get a `Illuminate\Http\UploadedFile` instance somewhere? So we can use all the handy methods there?

**Reply by ryangjchandler** [reply]
It might be possible to add an `->afterStateUpdated` callback to the field and then use `$set` to upload the value of another field. Not sure if the `$state` in the `afterStateUpdated` callback will be the `UploadedFile` or not though, haven't actually tried it.

**Reply by danharrin** [reply]
yeah, `$state` is an instance of `TemporaryUploadedFile` in `afterStateUpdated()`
